src/queries/__tests__/host-component-names.breaking.test.tsx

```diff
@@ -0,0 +1,143 @@
+  configure({
+    hostComponentNames: { text: 'wrongName', textInput: 'wrongName' },
+  });
+
+  render(<Text>hello</Text>);
+
+  expect(() => screen.getByText('hello')).toThrowErrorMatchingInlineSnapshot(`
+      "Unable to find an element with text: hello
+
+      You configuration contains invalid host component names. This can happen if you use a version of React Native that is not compatible with your version of @testing-library/react-native
+
+      To fix this add following line in a Jest setup file :
+
+      configure({ 
+        hostComponentNames: { 
+          text: 'Text',
+          textInput: 'TextInput',
+        }
+      });"
+    `);
+});
+
+test('suggests appropriate config when a getByDisplayValue query fails and wrong component names are used', () => {
+  configure({
+    hostComponentNames: { text: 'wrongName', textInput: 'wrongName' },
+  });
+
+  render(<TextInput value="value" />);
+
+  expect(() => screen.getByDisplayValue('value'))
+    .toThrowErrorMatchingInlineSnapshot(`
+      "Unable to find an element with displayValue: value
+
+      You configuration contains invalid host component names. This can happen if you use a version of React Native that is not compatible with your version of @testing-library/react-native
+
+      To fix this add following line in a Jest setup file :
+
+      configure({ 
+        hostComponentNames: { 
+          text: 'Text',
+          textInput: 'TextInput',
+        }
+      });"
+    `);
+});
+
+test('suggests appropriate config when a getByPlaceholderText query fails and wrong component names are used', () => {
+  configure({
+    hostComponentNames: { text: 'wrongName', textInput: 'wrongName' },
+  });
+
+  render(<TextInput placeholder="placeholder" />);
+
+  expect(() => screen.getByPlaceholderText('placeholder'))
+    .toThrowErrorMatchingInlineSnapshot(`
+      "Unable to find an element with placeholder: placeholder
+
+      You configuration contains invalid host component names. This can happen if you use a version of React Native that is not compatible with your version of @testing-library/react-native
+
+      To fix this add following line in a Jest setup file :
+
+      configure({ 
+        hostComponentNames: { 
+          text: 'Text',
+          textInput: 'TextInput',
+        }
+      });"
+    `);
+});
+
+test('does not suggest to change config when a query fails but the correct component names are used', () => {
+  render(<Text>bonjour</Text>);
+
+  expect(() => screen.getByText('hello')).toThrowErrorMatchingInlineSnapshot(
+    `"Unable to find an element with text: hello"`
+  );
+});
+
+test('does not update screen object when running detection', async () => {
+  const TestComponent = () => {
+    const [shouldDisplayHello, setShouldDisplayHello] = useState(false);
+
+    useEffect(() => {
+      setTimeout(() => {
+        setShouldDisplayHello(true);
+      }, 500);
```

Comment:
fake timers are not enabled by default in this repo so your test will take quite some time to run. We could either reduce the time in the timeout and/or active fake timers for this test

Reply:
Oh good catch ! I've activated fake timers for this test only